### ISSUE TITLE
Get experiment 27 working again with the latest code.

### ISSUE THREAD

**Issue by JackKelly** [openclimatefix/power_perceiver#153]
- [x] #157
- [x] #158
- [x] Check that the `satellite_processor` and `query_generators` all make use of the modality-specific sun position
- [x] Use `hrvsatellite_predicted.detach()` in SatelliteTransformer.
- [x] Reshape so each timestep is seen as a separate example in `HRVSatelliteProcessor`
- [x] Remove `random_timestep_indexes` from plotting & validation code
- [x] Use the t0_idx provided by the data loader in the model code.
- [x] #159
- [x] #161
